### PR TITLE
[Quant] Use the true src zero point to query and create conv pd

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #91155
* #91154
* #91153
* #91152
* __->__ #90818
* #90364

**Summary**
Previously, we use `DNNL_RUNTIME_S32_VAL` as the `zero point` for `src` in both weight prepack and convolution forward to ensure the same block format of weight is used. The problem is `DNNL_RUNTIME_S32_VAL` may query out a different block format weight comparing with the true `zero point` for `src`. It makes oneDNN convolution into `jit` path instead of `brgconv` path. Here we will use the true `zero point` for `src` to create pd and make reorder if it's a different block format weight as weight prepack generated.  

**Test Plan**
```
python -m pytest quantization/core/test_quantized_op.py::TestQuantizedConv::test_conv_transpose_reorder_issue_onednn
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Comment by leslie-fang-intel** [comment]
@Xia-Weiwen Please help to take a look.

File: aten/src/ATen/native/quantized/cpu/qconv.cpp [hunk 1]
```diff
@@ -1316,7 +1316,7 @@ at::Tensor PackedConvWeightsOnednn<kSpatialDim>::apply_impl(
     op_attr = kReluFused ? ideep::attr_t::fuse_relu() : ideep::attr_t();
   }
   // Since src zero point is unknown, set runtime value here
-  op_attr.set_zero_points(DNNL_ARG_SRC, ideep::utils::tensor_zp_mask(1), {DNNL_RUNTIME_S32_VAL});
+  op_attr.set_zero_points(DNNL_ARG_SRC, ideep::utils::tensor_zp_mask(1), src_zero_points);
```
Comment (jgong5) [inline aten/src/ATen/native/quantized/cpu/qconv.cpp:1319]: Are we use src zero point to initialize the primitive in the cache or use `DNNL_RUNTIME_S32_VAL` instead? If it is the latter, the primitive cache won't take effect any longer?

Comment (Xia-Weiwen) [reply aten/src/ATen/native/quantized/cpu/qconv.cpp:1319]: > Are we use src zero point to initialize the primitive in the cache or use `DNNL_RUNTIME_S32_VAL` instead? If it is the latter, the primitive cache won't take effect any longer?

We used to use runtime value to create primitive and put it in cache. This PR uses true values of src zero point to create a primitive.
Src zero point is part of the cache key. If src zero point changes, cache misses. It has always been this behavior, not introduced by this PR.

**Review by Xia-Weiwen — APPROVED** [review]
LGTM
BTW, `try_reorder` is no longer needed after this PR https://github.com/pytorch/pytorch/pull/90354 is landed which uses new ideep API. The `try_reorder` behavior is implemented inside ideep.

File: aten/src/ATen/native/quantized/cpu/qconv.cpp (hunk 1 above)
Comment (jgong5) [reply aten/src/ATen/native/quantized/cpu/qconv.cpp:1319]: OK. If we were using `DNNL_RUNTIME_S32_VAL` previously, the src zero point shouldn't be part of the key? Anyway, the change looks good then.

**Review by jgong5 — APPROVED** [review]

**Comment by leslie-fang-intel** [comment]
> LGTM BTW, `try_reorder` is no longer needed after this PR #90354 is landed which uses new ideep API. The `try_reorder` behavior is implemented inside ideep.

The change of `try_reorder` has been removed since #90354 is landed.

File: test/quantization/core/test_quantized_op.py [hunk 2]
```diff
@@ -5777,6 +5777,27 @@ def test_conv_reorder_issue_onednn(self):
             qx = torch.quantize_per_tensor(x, scale=1.0, zero_point=0, dtype=torch.quint8)
             # The following should pass when input shape is changed
             torch.ops.quantized.conv2d(qx, w_packed, output_scale=1.0, output_zero_point=0)
+        # conv_transposed part
```
Comment (jerryzh168) [inline test/quantization/core/test_quantized_op.py:5780]: maybe split this into a separate test?

Comment (leslie-fang-intel) [reply test/quantization/core/test_quantized_op.py:6235]: Thanks for the comment. Put it into a separate test.

**Comment by leslie-fang-intel** [comment]
@jerryzh168 I have changed according to your comments. Could you help to take a look of this PR again?

**Comment by leslie-fang-intel** [comment]
Hi @jerryzh168 Is there any other comments to this PR? Could you help to take a look again?

File: test/quantization/core/test_quantized_op.py [hunk 3]
```diff
@@ -6233,6 +6233,30 @@ def test_conv_reorder_issue_onednn(self):
             # The following should pass when input shape is changed
             torch.ops.quantized.conv2d(qx, w_packed, output_scale=1.0, output_zero_point=0)
 
+    def test_conv_transpose_reorder_issue_onednn(self):
+        if 'onednn' not in supported_qengines:
+            return
```
Comment (jerryzh168) [inline test/quantization/core/test_quantized_op.py:6238]: I think you can use: https://github.com/pytorch/pytorch/blob/master/test/quantization/core/test_quantized_op.py#L3992

**Review by jerryzh168 — APPROVED** [review]

File: test/quantization/core/test_quantized_op.py (hunk 3 above)
Comment (leslie-fang-intel) [reply test/quantization/core/test_quantized_op.py:6238]: Good suggestions. Changed it.

**Pull request closed without merging**